Temporary fix for:

dyld: lazy symbol binding failed: Symbol not found: _SCDynamicStoreCopyProxies
  Referenced from: /opt/local/lib/libcurl.4.dylib
  Expected in: flat namespace

due to failure to properly identify macOS when using the macOS 10.11 SDK or
earlier. This patch is only tested on macOS so far. The upstream fix, which
needs to accommodate iOS, tvOS, watchOS, and every other OS may be different.

https://github.com/curl/curl/issues/11502
--- configure.orig	2023-07-18 04:55:46.000000000 -0500
+++ configure	2023-07-24 15:22:39.000000000 -0500
@@ -21707,7 +21707,7 @@
 int main (void)
 {
 
-#if (TARGET_OS_OSX)
+#if TARGET_OS_MAC && !(defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)
       return 0;
 #else
 #error Not a macOS
--- lib/curl_setup.h.orig	2023-07-18 06:46:01.000000000 -0500
+++ lib/curl_setup.h	2023-07-24 15:22:52.000000000 -0500
@@ -258,7 +258,7 @@
 #if defined(__APPLE__) && !defined(USE_ARES)
 #include <TargetConditionals.h>
 #define USE_RESOLVE_ON_IPS 1
-#  if !defined(TARGET_OS_OSX) || TARGET_OS_OSX
+#  if TARGET_OS_MAC && !(defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)
 #    define CURL_OSX_CALL_COPYPROXIES 1
 #  endif
 #endif
--- lib/macos.c.orig	2023-07-18 04:46:48.000000000 -0500
+++ lib/macos.c	2023-07-24 15:23:43.000000000 -0500
@@ -26,7 +26,7 @@
 
 #if defined(__APPLE__)
 
-#if !defined(TARGET_OS_OSX) || TARGET_OS_OSX
+#if TARGET_OS_MAC && !(defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)
 
 #include <curl/curl.h>
 
--- lib/macos.h.orig	2023-07-18 04:46:48.000000000 -0500
+++ lib/macos.h	2023-07-24 15:23:16.000000000 -0500
@@ -25,7 +25,7 @@
  ***************************************************************************/
 #include "curl_setup.h"
 
-#if defined(__APPLE__) && (!defined(TARGET_OS_OSX) || TARGET_OS_OSX)
+#if defined(__APPLE__) && TARGET_OS_MAC && !(defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)
 
 CURLcode Curl_macos_init(void);
 
